xlive/Blam/Engine/kablam/kablam_log.cpp

```diff
@@ -0,0 +1,68 @@
+#include "stdafx.h"
+#include "kablam_log.h"
+
+#include "kablam.h"
+
+/* typedefs */
+
+typedef void(__cdecl* kablam_log_t)(const char*, ...);
+
+/* globals */
+
+static kablam_log_t p_kablam_log;
+
+/* prototypes */
+
+static void __cdecl kablam_log(const char* format, ...);
+
+/* public code */
+
+void kablam_log_apply_patches(void)
+{
+	// Detour because the base function doesn't work????
+	DETOUR_ATTACH(p_kablam_log, Memory::GetAddress<kablam_log_t>(0x0, 0x1D8C0), kablam_log);
+	return;
+}
+
+void kablam_log_initialize(void)
+{
+	wchar_t log_name[MAX_PATH] = L"reports\\";
+
+	// Make sure reports folder exists
+	CreateDirectoryW(log_name, NULL);
+
+	const wchar_t* instance_name = kablam_shell_argument_get(L"-instance:");
+	if (instance_name)
+	{
+		// Make sure instance folder exists
+		ustrncat(log_name, instance_name, NUMBEROF(log_name));
+		ustrncat(log_name, L"\\", NUMBEROF(log_name));
+		CreateDirectoryW(log_name, NULL);
+	}
+
+	ustrncat(log_name, L"kablam.txt", NUMBEROF(log_name));
+
+	*Memory::GetAddress<bool*>(0x0, 0x4904E8) = true;	// Set logging to true
```

Comment:
This should probably not be enforced as always on, probably should be a setting

Reply:
added -log parameter to enable it